### PR TITLE
API provides functionality for user to see their current order

### PR DESCRIPTION
**Pull request by PaulineBA**
[PT LINK](https://www.pivotaltracker.com/story/show/172641999)
### User Story
```As a restaurant API
In order to give the user a way to see their current order
I would like to make sure to open up a endpoint for that and make sure that a user can only have one current order
```
## Changes proposed in this pull request:
* We have added an end point to save order
## What I have learned working on this feature:
* I have learned about relations between order, order_items and user

### CODE REVIEW

File: spec/models/order_item_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe OrderItem, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (faraznaeem) [inline spec/models/order_item_spec.rb]: missing tests

File: spec/models/order_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Order, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (faraznaeem) [inline spec/models/order_spec.rb]: missing tests

File: spec/requests/api/v1/functionality_to_see_current_order_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+RSpec.describe Api::V1::OrdersController, type: :request do 
+let!(:product_1) {create(:product, title: 'Pizza', id: 1)}
+let!(:product_2) {create(:product, title: 'Hamburger', id: 2)}
+  it "successfully" do 
+    post '/api/v1/orders', params: {id:product_1}
```
Comment (faraznaeem) [inline spec/requests/api/v1/functionality_to_see_current_order_spec.rb]: describe block, indentation etc

File: spec/requests/api/v1/user_can_have_order_items_in_his_order_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+RSpec.describe Order, type: :model do 
+  describe 'order should have item' do 
+    it { is_expected.to have_many :order_items }
+  end
+end
+RSpec.describe OrderItem, type: :model do
```
Comment (faraznaeem) [inline spec/requests/api/v1/user_can_have_order_items_in_his_order_spec.rb]: split this into multiple files

**Review by faraznaeem — CHANGES_REQUESTED** [review]
Title and user story is off.

File: db/schema.rb [hunk 5]
```diff
@@ -10,11 +10,25 @@
...
+    t.bigint "product_id", null: false
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["order_id"], name: "index_order_items_on_order_id"
+    t.index ["product_id"], name: "index_order_items_on_product_id"
+  end
+
+  create_table "orders", force: :cascade do |t|
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
```
Comment (erikbjoern) [inline db/schema.rb:29]: Since order_items belong to order, shouldn't orders have some column for that?

**Review by erikbjoern — COMMENTED** [review]
Left one comment about db table 'orders'.

File: db/schema.rb (hunk 5 above)
Comment (jysmys) [reply db/schema.rb:29]: I have tried in rails console and that worked fine i think. But would be good if you try as well!

File: spec/models/order_spec.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+RSpec.describe Order, type: :model do 
+  describe 'order should have item' do
```
Comment (oliverochman) [inline spec/models/order_spec.rb]: This should just say "Associations"

File: spec/requests/api/v1/functionality_to_see_current_order_spec.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    expect(JSON.parse(response.body)['message']).to eq 'The product has been added to your order'
```
Comment (oliverochman) [inline spec/requests/api/v1/functionality_to_see_current_order_spec.rb]: extract `JSON.parse(response.body)` to a helper

File: spec/models/order_item_spec.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+RSpec.describe OrderItem, type: :model do 
+  describe 'Database table' do
+    it { is_expected.to have_db_column :order_id }
+    it { is_expected.to have_db_column :product_id }
```
Comment (oliverochman) [inline spec/models/order_item_spec.rb:4]: you need to test the factory bot here

File: spec/models/order_spec.rb (hunk 6 above)
Comment (oliverochman) [reply spec/models/order_spec.rb]: you need to test the factory bot here as well

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/functionality_to_see_current_order_spec.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    post '/api/v1/orders', 
+    params: {
+      id:product_1,
+    }
+    @order_id = JSON.parse(response.body)["order_id"]
+  end
+  
+  describe 'POST /order' do
+    it "successfully" do       
+      expect(JSON.parse(response.body)['message']).to eq 'The product has been added to your order'
```
Comment (oliverochman) [inline spec/requests/api/v1/functionality_to_see_current_order_spec.rb]: i need you to extract `JSON.parse(response.body)` to a helper method

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/factories/order_items.rb [hunk 10]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
```
Comment (oliverochman) [inline spec/factories/order_items.rb:1]:
```
FactoryBot.define do
  factory :order_items, class: OrderItem do
    order
    product
  end
end
```

Comment (oliverochman) [reply spec/factories/order_items.rb:1]: Have it like this instead

File: spec/models/order_item_spec.rb [hunk 11]
```diff
@@ -0,0 +1,24 @@
+RSpec.describe OrderItem, type: :model do 
+  let!(:product) { create(:product,
```
Comment (oliverochman) [inline spec/models/order_item_spec.rb]: if you change the factorybot you wont need line 2 to 7

File: spec/models/order_spec.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
```
Comment (oliverochman) [inline spec/models/order_spec.rb]: please change the message for the describe block to "Associations"

File: spec/requests/api/v1/functionality_to_see_current_order_spec.rb (hunk 9 above)
Comment (oliverochman) [reply spec/requests/api/v1/functionality_to_see_current_order_spec.rb]: like we did in the cooper challenge

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**